Add umbraco commerce product feed package (#6007)
umbraco/UmbracoDocs: commerce-add-ons/packages/product-feeds/extending.md

Reviewer: I would suggest adding an example to why you would extend the plugin. In what use case would you need to extend the plugin?

Author: I added an example.

# Extending the plugin

## Add a custom property value extractor

> When a simple node-name-to-property-alias mapping does not suit your need, you can create your own property value extractor to... *extract* the value from the property yourself.

Reviewer 2: ```suggestion
When a default node-name-to-property-alias mapping does not suit your need, you can create a property value extractor to *extract* the value from the property.
```

> Most of the time, you just need to create a new implementation of `ISingleValuePropertyExtractor` ~~or rarely `IMultipleValuePropertyExtractor`~~.

Reviewer 2: ```suggestion
You can often create a new implementation of `ISingleValuePropertyExtractor` or rarely `IMultipleValuePropertyExtractor`.
```

This plugin uses [Collection Builder pattern](https://docs.umbraco.com/umbraco-cms/implementation/composing#example-modifying-collections) which is commonly used in Umbraco. You can use these two extension methods during application initialization to add your value extractors.

```c#
// IUmbracoBuilder builder;
builder.SingleValuePropertyExtractors()
    .Append<DefaultSingleValuePropertyExtractor>()
    .Append<DefaultGoogleAvailabilityValueExtractor>()
    .Append<DefaultMediaPickerPropertyValueExtractor>();

builder.MultipleValuePropertyExtractors()
    .Append<DefaultMultipleMediaPickerPropertyValueExtractor>();
```

> Afterwards, your extractor name should show up in the dropdown under `Property And Node Mapping` section.
> ![property value extractor dropdown](./media/property-value-extractor-dropdown.png)

Reviewer 2: ```suggestion
Afterwards, your extractor name should show up in the dropdown under `Property And Node Mapping` section.

![property value extractor dropdown](./media/property-value-extractor-dropdown.png)
```

> # Extending the plugin

Reviewer 2: ```suggestion
---
description: >-
 Learn how to extend the plugin by adding a custom property value extractor.
---

# Extending the plugin

In this article, you can find an example of extending the plugin by adding a custom property value extractor.
```
